doc/sphinx/addendum/type-classes.rst

```diff
@@ -1,578 +1,616 @@
+   during type class resolution. It is useful when some constants
+   prevent some unifications and make resolution fail. It is also useful
+   to declare constants which should never be unfolded during
+   proof-search, like fixpoints or anything which does not look like an
+   abbreviation. This can additionally speed up proof search as the
+   typeclass map can be indexed by such rigid constants (see
+   `8.9.1`_). By default, all constants and local variables are
+   considered transparent. One should take care not to make opaque any
+   constant that is used to abbreviate a type, like
+
+::
+
+   relation A := A -> A -> Prop.
+
+This is equivalent to ``Hint Transparent, Opaque ident : typeclass_instances``.
+
+
+Set Typeclasses Dependency Order
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This option (on by default since 8.6) respects the dependency order
+between subgoals, meaning that subgoals which are depended on by other
+subgoals come first, while the non-dependent subgoals were put before
+the dependent ones previously (Coq v8.5 and below). This can result in
+quite different performance behaviors of proof search.
+
+
+Set Typeclasses Filtered Unification
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 This option, available since Coq 8.6 and off by default, switches the
 hint application procedure to a filter-then-unify strategy. To apply a
-hint, we first check that the goal \emph{matches} syntactically the
+hint, we first check that the goal *matches* syntactically the
 inferred or specified pattern of the hint, and only then try to
-\emph{unify} the goal with the conclusion of the hint. This can
-drastically improve performance by calling unification less often,
-matching syntactic patterns being very quick. This also provides more
-control on the triggering of instances.  For example, forcing a constant
-to explicitely appear in the pattern will make it never apply on a goal
+*unify* the goal with the conclusion of the hint. This can drastically
+improve performance by calling unification less often, matching
+syntactic patterns being very quick. This also provides more control
+on the triggering of instances. For example, forcing a constant to
+explicitely appear in the pattern will make it never apply on a goal
 where there is a hole in that place.
 
-\subsection{\tt Set Typeclasses Limit Intros}
-\optindex{Typeclasses Limit Intros}
 
-This option (on by default) controls the ability to
-apply hints while avoiding (functional) eta-expansions in the generated
-proof term. It does so by allowing hints that conclude in a product to
-apply to a goal with a matching product directly, avoiding an
-introduction. \emph{Warning:} this can be expensive as it requires
-rebuilding hint clauses dynamically, and does not benefit from the
-invertibility status of the product introduction rule, resulting in
-potentially more expensive proof-search (i.e. more useless
-backtracking).
+Set Typeclasses Legacy Resolution
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
This option was removed in the 8.8 manual. You obviously followed the advice of starting off the 8.7 HTML manual. Well, that was bad advice ;-)